src/main/kotlin/com/the_chance/data/services/CategoryService.kt

```diff
@@ -0,0 +1,65 @@
+package com.the_chance.data.services
+
+import com.the_chance.data.models.Category
+import com.the_chance.data.tables.CategoriesTable
+import org.jetbrains.exposed.sql.*
+
+class CategoryService(
+    private val database: Database
+) : BaseService(database) {
+
+    suspend fun create(categoryName: String, categoryImage: String): Category = dbQuery {
+        val newCategory = CategoriesTable.insert {
+            it[name] = categoryName
+            it[image] = categoryImage
+            it[isDeleted] = false
+        }
+        Category(
+            id = newCategory[CategoriesTable.id].value,
+            name = newCategory[CategoriesTable.name].toString(),
+            image = newCategory[CategoriesTable.image].toString(),
+        )
+    }
+
+    suspend fun getAllCategories(): List<Category> {
+        return dbQuery {
+            CategoriesTable.select { CategoriesTable.isDeleted eq false }.map { resultRow ->
+                Category(
+                    id = resultRow[CategoriesTable.id].value,
+                    name = resultRow[CategoriesTable.name].toString(),
+                    image = resultRow[CategoriesTable.image].toString(),
+                )
+            }
+        }
+    }
+
+    suspend fun remove(categoryId: Int?): Boolean = dbQuery {
+        val category = CategoriesTable.select { CategoriesTable.id eq categoryId }.singleOrNull()
+        val isCategoryDeleted = CategoriesTable.select { CategoriesTable.isDeleted eq false }.singleOrNull()
```

Comment:
Since the category already appears for the user, there is no need to check it's deleted to delete it again!! + its access for DB for no real need